protocol-tokio/src/protocol/inbound_stream.rs

```diff
@@ -58,6 +58,7 @@ pub enum Inbound {
 pub struct InboundStream<T> {
     stream: SplitStream<nt::Connection<T>>,
     state: Arc<Mutex<ConnectionState>>,
+    node_received: bool,
```

Comment:
I don't see the value of checking this. We don't need this at all because if the next message ends up in the `InboundStream::forward_message` it will fail to find the node_id and will return an error. The need for the `node_received` is already enforced by design of the protocol itself.